stakingModel/test/UBIScheme.e2e.test.js

```diff
@@ -0,0 +1,192 @@
+    const dao_file = await fse.readFile("../releases/deployment.json", "utf8");
+    const staking_deployment = await JSON.parse(staking_file);
+    const dao_deployment = await JSON.parse(dao_file);
+    const staking_addresses = staking_deployment[NETWORK];
+    const dao_addresses = dao_deployment[NETWORK];
+    avatarAddress = dao_addresses.Avatar;
+    ubiBridgeRecipient = staking_addresses.UBIScheme;
+    dai = await DAIMock.at(staking_addresses.DAI);
+    cDAI = await cDAIMock.at(staking_addresses.cDAI);
+    simpleStaking = await SimpleDAIStaking.at(staking_addresses.DAIStaking);
+    goodReserve = await GoodReserve.at(staking_addresses.Reserve);
+    goodFundManager = await GoodFundsManager.at(staking_addresses.FundManager);
+    marketMaker = await MarketMaker.at(staking_addresses.MarketMaker);
+    contribution = await ContributionCalculation.at(staking_addresses.Contribution);
+    controller = await Controller.at(dao_addresses.Controller);
+    ubi = await UBI.at(staking_addresses.UBIScheme);
+    firstClaimPool = await FirstClaimPool.at(staking_addresses.FirstClaimPool);
+    identity = await Identity.at(dao_addresses.Identity);
+    goodDollar = await GoodDollar.at(dao_addresses.GoodDollar);
+    registrar = await SchemeRegistrar.at(dao_addresses.SchemeRegistrar);
+    absoluteVote = await AbsoluteVote.at(dao_addresses.AbsoluteVote);
+    deploy_settings = await fse.readFile("../migrations/deploy-settings.json", "utf8");
+    // schemes
+    setReserve = await FundManagerSetReserve.new(avatarAddress, goodFundManager.address, goodReserve.address);
+    addMinter = await AddMinter.new(avatarAddress, goodReserve.address);
+  });
+  
+  it("should start the pool", async () => {
+    let isActive = await firstClaimPool.isActive();
+    if (!isActive)
+      await firstClaimPool.start();
+    isActive = await firstClaimPool.isActive();
+    expect(isActive).to.be.true;
+  });
+
+  it("should be able to set the reserve as a minter", async () => {
+    let isMinter = await goodDollar.isMinter(goodReserve.address);
+    if (!isMinter) {
+      await proposeAndRegister(addMinter.address, 
+        registrar, proposalId, absoluteVote, avatarAddress, founder);
+      await addMinter.addMinter();
+    }
+    let amount = 1e8;
+    await dai.mint(web3.utils.toWei("1000", "ether"));
+    dai.approve(cDAI.address, web3.utils.toWei("1000", "ether"));
+    await cDAI.mint(web3.utils.toWei("1000", "ether"));
+    await cDAI.approve(goodReserve.address, amount);
+    await goodReserve.buy(cDAI.address, amount, 0);
+    let gdbalance = await goodDollar.balanceOf(founder);
+    await goodDollar.transfer(firstClaimPool.address, (Math.floor(gdbalance.toNumber() / 2)).toString());
+    await goodDollar.transfer(ubi.address, (Math.floor(gdbalance.toNumber() / 2)).toString());
+    expect((await goodDollar.balanceOf(ubi.address)).toNumber()).to.be.gt(0);
+    expect((await goodDollar.balanceOf(firstClaimPool.address)).toNumber()).to.be.gt(0);
```

Comment:
what does testing transfer to ubi and firstclaim has to do with this test?
you want to test reserve can mint. the test finished when the buyer got his GD
you can test GD total cap has increased

Reply:
firstclaim contract should has funds.
db43a7633912fed7aa89a7ff580fbe5238d650b5